### PR TITLE
GEOWAVE-348

### PR DESCRIPTION
**Pull request by rwgdrummer**

### CODE REVIEW

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java [hunk 1]
```diff
@@ -138,7 +139,7 @@ public Serializable get(
 		try {
 			convertedValue = converter.convert(value);
 		}
-		catch (Exception e) {
+		catch (final Exception e) {
```
Comment (chrisbennight) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java]: If we re-throw an new exception we should pass the old exception to the constructor of the new one so the stack trace is maintained (looks like it wasn't changed in this pull request - just noticed it here)

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/kmeans/mapreduce/KMeansMapReduce.java [hunk 2]
```diff
@@ -53,7 +60,7 @@
 public class KMeansMapReduce
 {
 
-	protected static final Logger LOGGER = Logger.getLogger(KMeansReduce.class);
+	protected static final Logger LOGGER = LoggerFactory.getLogger(KMeansReduce.class);
```
Comment (chrisbennight) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/kmeans/mapreduce/KMeansMapReduce.java]: Class should be KMeansMapReduce.class in the getLogger method

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/mapreduce/JobContextConfigurationWrapper.java [hunk 3]
```diff
@@ -3,16 +3,15 @@
...
+import org.slf4j.LoggerFactory;
 
 public class JobContextConfigurationWrapper implements
 		ConfigurationWrapper
 {
-	protected static final Logger LOGGER = Logger.getLogger(JobContextConfigurationWrapper.class);
+	protected static final Logger LOGGER = LoggerFactory.getLogger(JobContextConfigurationWrapper.class);
 
 	private final JobContext context;
 	private Logger logger = LOGGER;
```
Comment (chrisbennight) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/mapreduce/JobContextConfigurationWrapper.java]: Wouldn't let me comment on it, but just below this
```
	public JobContextConfigurationWrapper(
			final JobContext context,
			final Logger logger ) 
```

What's the rationale here for passing around a logger instance?

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/partitioners/AdapterBasedPartitioner.java [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+ * The user provides the distances per dimension. It us up to the user to
+ * convert geographic distance into distance in degrees per longitude and
+ * latitude.
+ * 
+ * 
+ */
+public class AdapterBasedPartitioner extends
+		AbstractPartitioner<AdapterDataEntry> implements
+		Partitioner<AdapterDataEntry>,
+		Serializable
```
Comment (chrisbennight) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/partitioners/AdapterBasedPartitioner.java]: Is this class ever going to get serialized?   If so, do we need to set defaults for the transients, or serialize them in a readObject/writeObject?

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/GeometryCalculations.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+	 * @param unit
+	 * @param coordinate
+	 * @return
+	 */
+	public List<Geometry> buildSurroundingGeometries(
+			final double[] distances,
+			final Unit<Length> unit,
+			Coordinate coordinate ) {
+		List<Geometry> geos = new LinkedList<Geometry>();
+		GeodeticCalculator geoCalc = new GeodeticCalculator();
```
Comment (chrisbennight) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/GeometryCalculations.java]: I move we don't use the geotools geodetic calculator;  if you do you need to guard / give up with antipodal pairs.  See: https://github.com/chrisbennight/geodesy - I think GeographicLib has a pure java implementation that's better on pretty much all counts.   It's not in maven central, but is MIT licensed

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/mapreduce/JobContextConfigurationWrapper.java (hunk 3 above)
Comment (rwgdrummer) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/mapreduce/JobContextConfigurationWrapper.java]: So the context of the request for a property is logged appropriately.

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java (hunk 1 above)
Comment (rwgdrummer) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java]: k

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/kmeans/mapreduce/KMeansMapReduce.java (hunk 2 above)
Comment (rwgdrummer) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/kmeans/mapreduce/KMeansMapReduce.java]: k

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/partitioners/AdapterBasedPartitioner.java (hunk 4 above)
Comment (rwgdrummer) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/partitioners/AdapterBasedPartitioner.java]: The issue here is AdapterStore is not serializable.  I think there is a more appropriate way to handle this now without forcing an interface change to AdapterStore.  Will adjust without some treatment now and perhaps revisit later.

File: geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/GeometryCalculations.java (hunk 5 above)
Comment (rwgdrummer) [inline geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/GeometryCalculations.java]: Agreed.  Talking to Rich, I would like to tackle this as a separate ticket (immediately).  Thoughts?

**Pull request merged**